To-do list/Controllers/TaskListViewController.swift

```diff
@@ -44,42 +45,45 @@ class TaskListViewController: UIViewController, TaskTableViewCellDelegate {
             
             
             switch currentStatus {
-            case "in progress":
-                currentStatus = "done"
-                print("current status: \(currentStatus)")
+            case .inProgress:
+                currentStatus = .done
                 tasks[chosenIndex.row].status = currentStatus
                 tasksTableView.reloadData()
-                users[userIndex].task?[chosenIndex.row].status = currentStatus
-                UserDefaultsManager.shared.setValueForUser(value: users)
-            case "done":
-                currentStatus = "to do"
-                print("current status: \(currentStatus)")
+            case .done:
+                currentStatus = .todo
                 tasks[chosenIndex.row].status = currentStatus
                 tasksTableView.reloadData()
-                users[userIndex].task?[chosenIndex.row].status = currentStatus
-                UserDefaultsManager.shared.setValueForUser(value: users)
-            case "to do":
-                currentStatus = "in progress"
-                print("current status: \(currentStatus)")
+            case .todo:
+                currentStatus = .inProgress
                 tasks[chosenIndex.row].status = currentStatus
                 tasksTableView.reloadData()
-                users[userIndex].task?[chosenIndex.row].status = currentStatus
-                UserDefaultsManager.shared.setValueForUser(value: users)
             default:
                 break
             }
         }
     }
+    
+    func sendTaskDetails(_: TaskDetailsViewController, didCreateUpdate task: Task) {
+        if choosenTableViewCell == false {
+            tasks.append(task)
+            tasksTableView.reloadData()
+        } else {
+            tasks[taskIndex] = task
+            tasksTableView.reloadData()
```

Comment:
every task should have a unique Int id. This is how you can distinguish if task already exist in the Array, or it is new task returned.

Reply:
added id to Task and used it for replacing  var taskIndex.